cli/cmd/install.go

```diff
@@ -259,3 +165,75 @@ func checkIstioInstallation() error {
 
 	return nil
 }
+
+func checkInput(params installUpgradeParams, values map[string]interface{}) error {
+	isOpenShiftEnabled := isOpenShiftEnabled(values)
+	isContinuousDeliveryEnabled := isContinuousDeliveryEnabled(values)
+
+	platformManager := platform.NewPlatformManager(isOpenShiftEnabled)
+	if !mocking {
+		if err := platformManager.CheckRequirements(); err != nil {
+			return err
+		}
+	}
+
+	if params.ConfigFilePath != nil && *params.ConfigFilePath != "" {
+		// Config was provided in form of a file
+		if err := platformManager.ParseConfig(*params.ConfigFilePath); err != nil {
+			return err
+		}
+
+		// Check whether the authentication at the cluster is valid
+		if err := platformManager.CheckCreds(); err != nil {
+			return err
+		}
+	} else {
+		err := platformManager.ReadCreds()
+		if err != nil {
+			return err
+		}
+	}
+
+	// check if Kubernetes server version is compatible (except OpenShift)
+	if !isOpenShiftEnabled {
+		if err := kube.CheckKubeServerVersion(KubeServerVersionConstraints); err != nil {
+			logging.PrintLog(err.Error(), logging.VerboseLevel)
+			logging.PrintLog("See https://keptn.sh/docs/"+keptnReleaseDocsURL+"/operate/k8s_support/ for details.", logging.VerboseLevel)
+			return fmt.Errorf("Failed to check kubernetes server version: %w", err)
+		}
+	}
+
+	// check if istio-system namespace and istio-ingressgateway is available (full installation only)
+	if isContinuousDeliveryEnabled {
+		if err := checkIstioInstallation(); err != nil {
+			logging.PrintLog("Istio is required for values.continuous-delivery.enabled: true, "+
+				"but could not be found in your cluster in namespace istio-system.", logging.InfoLevel)
+			logging.PrintLog("Please install Istio as described "+
+				"in the Istio docs: https://istio.io/latest/docs/setup/getting-started/", logging.InfoLevel)
+		}
+	}
+
+	return nil
+}
+
+func isOpenShiftEnabled(values map[string]interface{}) (enabled bool) {
+	defer func() {
+		if recover() != nil {
+			// default chart value
+			enabled = false
+		}
+	}()
+	enabled = values["continuous-delivery"].(map[string]interface{})["openshift"].(map[string]interface{})["enabled"].(bool)
+	return
+}
```

Comment:
This seems a bit hacky.
I chose it instead of hardcoding the complete values file (when not provided by user).
Since here we're only hardcoding the default values that we need, instead of the complete values file.

Please let me know if there is a better approach.